kotskinds/apis/kots/v1beta1/helmchart_types.go

```diff
@@ -175,6 +176,88 @@ func (h *HelmChartSpec) GetHelmValues(values map[string]MappedChartValue) (map[s
 	return result, nil
 }
 
+func (h *HelmChartSpec) GetReplTmplValues(values map[string]MappedChartValue) (map[string]interface{}, error) {
+	newValues := make(map[string]interface{})
+
+	for k, v := range values {
+		value, err := h.getReplTmplValue(&v)
+		if err != nil {
+			continue
+		}
+
+		newValues[k] = value
+	}
+
+	return newValues, nil
+}
+
+func (h *HelmChartSpec) getReplTmplValue(value *MappedChartValue) (interface{}, error) {
+	if value.valueType == "children" {
+		result := map[string]interface{}{}
+		for k, v := range value.children {
+			built, err := h.getReplTmplValue(v)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to render child value at key %s", k)
+			}
+			result[k] = built
+		}
+		return result, nil
+	} else if value.valueType == "array" {
+		result := []interface{}{}
+		for _, v := range value.array {
+			built, err := h.getReplTmplValue(v)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to render array value")
+			}
+			result = append(result, built)
+		}
+		return result, nil
+	} else {
+		built, err := value.getBuiltValue()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to build value")
+		}
+		str, ok := built.(string)
+		if ok && (strings.HasPrefix(str, "repl{{") || strings.HasPrefix(str, "{{repl")) && strings.HasSuffix(str, "}}") {
```

Comment:
This check should be more like [this](https://github.com/replicatedhq/kots/blob/7990f0586fa5e7e6897182a6fd9b6f31b6ccf056/pkg/kotsutil/kots.go#L414) because templates can be anywhere in the string.